File: SCL/Datastruct/queue/模拟队列.py
if __name__ == "__main__":
    q = []
    n = int(input())
    for i in range(n):
        op = input().split()
        if op[0] == "push":
            q.append(op[1])
        elif op[0] == "pop":
                q = q[1:]
        elif op[0] == "empty":
            if q:
                print("NO")
            else:
                print("YES")
        else:
            print(q[0])

File: SCL/Datastruct/queue/模拟队列.cpp
#include <iostream>

using namespace std;

const int N = 1e5 + 10;

int que[N],hh,tt = -1;

//往队尾部插入
void push(int x){
    que[++tt] = x;
}

//弹出队列的头部
void pop(){
    hh++;
}

bool empty(){
    if(hh <= tt){
        return false;
    }
    return true;
}

int query(){
    return que[hh];
}

int main(){
    int n;
    cin >> n;
    while(n--){
        string op;
        cin >> op;
        int x;
        if(op == "push"){
            cin >> x;
            push(x);
        }else if (op == "pop"){
            pop();
        }else if(op == "empty"){
            
            cout << (empty()?"YES":"NO") << endl;
        }else{
            if (hh <= tt){
                cout << que[hh]<<endl;
            }
        }
    }
    return 0;
}